src/storage-client/src/controller.rs

```diff
@@ -1505,38 +1601,53 @@ where
     #[tracing::instrument(level = "debug", skip(self))]
     fn update_write_frontiers(&mut self, updates: &[(GlobalId, Antichain<Self::Timestamp>)]) {
         let mut read_capability_changes = BTreeMap::default();
-        let mut collections = BTreeMap::new();
-        let mut exports = vec![];
 
         for (id, new_upper) in updates.iter() {
-            if let Ok(_) = self.collection(*id) {
-                collections.insert(*id, Some(new_upper));
-            } else if let Ok(_) = self.export(*id) {
-                exports.push((id, new_upper));
-            } else {
-                panic!("Reference to absent collection");
-            }
-        }
+            if let Ok(collection) = self.collection_mut(*id) {
```

Comment:
So this is almost entirely besides the point of this PR, but -

For a few reasons (the "what to do with sink errors" question, cluster unification, etc.) I've been wondering whether it would make sense for a sink to have an output collection. It would be a slightly odd collection, in that it would never have any actual records. But it would give us somewhere to report errors - either just passing along the errors from the sink's input, or additional errors like an encoding failure. And it would help make code like this more uniform... we wouldn't need distinct paths for regular collections and exports.

I haven't actually worked through the details, so, grain of salt. Just a little galaxy brain for a Friday...

Reply:
Yeah, this is an interesting idea. The fact that we silently ignore errors from the input `err` collection has been bugging me a fair bit now... :see_no_evil:

Reply:
I love this idea! Many things fall into place like querying `EXPLAIN TIMESTAMP FOR SELECT * FROM sink` will tell you how far along that sink is